### PR TITLE
Add Solaris and FreeBSD support

### PR DESCRIPTION
**Pull request by mropert**

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -113,14 +115,52 @@ def run_in_src(self, command, show_output=False):
...
-            config_options_string = "-d no-asm -g3 -O0 -fno-omit-frame-pointer " \
-                                    "-fno-inline-functions" + config_options_string
+            config_options_string = " no-asm" + config_options_string
+            extra_flags = extra_flags + " -O0"
+            target_prefix = "debug-"
+            if self.settings.compiler in ["apple-clang", "clang", "gcc"]:
+                extra_flags = extra_flags + " -g3 -fno-omit-frame-pointer " \
+                                            "-fno-inline-functions"
+            else:
+                extra_flags = extra_flags + " -g"
```
Comment (lasote) [inline conanfile.py]: `-g` flag should be automatically appended by AutoToolsBuildEnvironment, right?

Comment (mropert) [inline conanfile.py]: I don't know, does it ? I admit I don't know that helper well...

Comment (lasote) [inline conanfile.py]: Why did you introduce it? What are the needed flags? -m32?

Comment (mropert) [inline conanfile.py]: Yes, I wanted conan to take care of the bitness flags depending on the `arch` settings

Comment (lasote) [inline conanfile.py]: Ok then, you can remove the `-g`

Comment (mropert) [inline conanfile.py]: Great, does it also take care of the -g3 on gcc/clang?

Comment (lasote) [inline conanfile.py]: Nope.

Comment (mropert) [inline conanfile.py]: Ok so I keep that part but remove the -g?

Comment (lasote) [inline conanfile.py]: Yes. Remove the "else"

**Pull request merged**